pika/adapters/blocking_connection.py

```diff
@@ -1496,12 +1496,12 @@ def add_on_return_callback(self, callback):
                         callback, self, method, properties, body))))
 
     def basic_consume(self,
-                      consumer_callback,
                       queue,
                       no_ack=False,
                       exclusive=False,
                       consumer_tag=None,
-                      arguments=None):
+                      arguments=None,
+                      callback=None):
```

Comment:
`callback=None` in the function signature suggests that `callback` is optional. However, the code and semantics dictate that it's required. This is likely confusing to the API user.

Reply:
Thanks for the feedback, I was wondering about that.